### PR TITLE
Add middleware for user-agent

### PR DESCRIPTION
**Pull request by xxbillouxx**

### CODE REVIEW

**Comment by xxbillouxx** [comment]
recheck

**Comment by xxbillouxx** [comment]
recheck

**Comment by xxbillouxx** [comment]
recheck

**Comment by xxbillouxx** [comment]
recheck

File: integration_tests/suite/base/test_users.py [hunk 1]
```diff
@@ -915,6 +915,7 @@ def test_post_full_user_no_error(
         group,
         switchboard,
     ) = generate_user_resources_bodies(group, switchboard)
+    agent = {}
```
Comment (pc-m) [inline integration_tests/suite/base/test_users.py:918]: oh shit, I did not anticipate a body that would be THAT small. When you know about the implementation it makes sense but as a user I would expect a boolean.

What would you think of allowing `"agent": true` that would be equivalent to `"agent": {}`
The following values would be possible.
true
{"firstname": "Nabil", "number": "666"}
null
and false I guess

@fblackburn1 Do you have an idea about this one?

**Review by pc-m — CHANGES_REQUESTED** [review]
api.yml and changelog missing

File: integration_tests/suite/base/test_users.py (hunk 1 above)
Comment (xxbillouxx) [reply integration_tests/suite/base/test_users.py:1005]: I think that a field having 2 differents types (dict and boolean) is ambiguous.
What's opinion @fblackburn1 ?

Comment (fblackburn1) [reply integration_tests/suite/base/test_users.py:1005]: Ooh well .. I agree with @xxbillouxx 
But understand that `agent: {}` may be not obvious.
Maybe a key helper `with_agent: true`? (only for the POST body, not for the returned body) - (and not required to be in this PR/scope)

Anyways, not a strong opinion, do what you want :smile:

Comment (pc-m) [reply integration_tests/suite/base/test_users.py:1005]: Ok then we are going to do whatever @xxbillouxx wants here :)
Also I'm going to take a screenshot of @fblackburn1 not having a strong opinion :rofl:

**Review by pc-m — APPROVED** [review]

**Pull request merged**